test/e2e/define/define.test.ts

```diff
@@ -5,17 +5,59 @@ describe('compiler.define', () => {
     files: __dirname,
   })
 
-  it('should render the magic variable on server side', async () => {
-    const res = await next.fetch('/')
-    const html = (await res.text()).replaceAll(/<!-- -->/g, '')
-    expect(html).toContain('Server value: foobar')
-    expect(html).toContain('Client value: foobar')
+  describe('compiler.define', () => {
+    it('should render the magic variable on server side', async () => {
+      const res = await next.fetch('/')
+      const html = (await res.text()).replaceAll(/<!-- -->/g, '')
+      expect(html).toContain('Server value: foobar')
+      expect(html).toContain('Client value: foobar')
+    })
+
+    it('should render the magic variable on client side', async () => {
+      const browser = await next.browser('/')
+      const text = await browser.elementByCss('body').text()
+      expect(text).toContain('Server value: foobar')
+      expect(text).toContain('Client value: foobar')
+    })
+
+    it('should render the magic expression on server side', async () => {
+      const res = await next.fetch('/')
+      const html = (await res.text()).replaceAll(/<!-- -->/g, '')
```

Comment:
This looks off. It's better to use cheerio here.

Example of how to use cheerio:

```js
const $ = await next.render$('/')
expect($('p').text()).toBe('hello world')
```

In this case you'd want to add an `id` or `className` to the specific item.